include/stream-client/connector/impl/connection_pool.ipp

```diff
@@ -108,7 +120,106 @@ bool base_connection_pool<Connector>::is_connected(boost::system::error_code& ec
 }
 
 template <typename Connector>
-void base_connection_pool<Connector>::watch_pool_routine()
+struct reconnection_strategy_greedy {
```

Comment:
Both `reconnection_strategy_greedy` and `reconnection_strategy_conservative` should be declared, described and implemented in separate file, maybe `/connector/pool_strategy.hpp` and the .ipp file. There probably should be an interface for the strategy to allow the user to implement its own strategy.

Reply:
Moved it to pool_strategy files and described a bit more.